### DISCUSSION TITLE
Virtual channels (Channels generated base on criterias)

### DISCUSSION THREAD

**Discussion by jespino** [mattermost/community#136] [Moonshot Ideas]
One example of this could be a virtual channel that is populated with search results, another option is a virtual channel that is populated with all the posts of a list of channels.

Also we can make the virtual channels more powerful and have things like "rss-feed" based channel, or google-group mailing list based channel, maybe another using bamboo to show the history of the organization changes (Who is joining, who is leaving, who is being promoted, who is taking PTO...). This is just a idea that can explode in a crazy amount of other ideas.

Of course this should be supported by the plugins system, and have the more general versions, like search-based ones directly in the core product.

**Comment by chetanyakan** [comment]
So, you mean just getting the posts list from a plugin/integration and having it show up on Mattermost like default channels.

Maybe we should have a different icon show up in front of virtual channels (different from public/private channel icons).

**Reply by jespino** [reply]
Yes the idea is that the post is generated on the fly from wherever, the simplest thing is something like search results, but there are other cool things that can be implemented

**Reply by chetanyakan** [reply]
Seems like a good idea for an RHS plugin.
Is there any advantage of modifying the channel view?
